src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/EvidenceUploadNotificationExternalTaskListener.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.reform.civil.service.tasklisteners;
+
+import org.camunda.bpm.client.ExternalTaskClient;
+import org.camunda.bpm.client.topic.TopicSubscriptionBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.civil.handler.tasks.EvidenceUploadCheckHandler;
+
+@Component
+public class EvidenceUploadNotificationExternalTaskListener {
+
+    private static final String TOPIC = "EVIDENCE_UPLOAD_CHECK";
+
+    @Autowired
+    private EvidenceUploadNotificationExternalTaskListener(EvidenceUploadCheckHandler evidenceUploadCheckHandler,
+                                                           ExternalTaskClient client) {
+        TopicSubscriptionBuilder subscriptionBuilder = client.subscribe(TOPIC);
+        subscriptionBuilder.handler(evidenceUploadCheckHandler).open();
+    }
```

Comment:
I can see why you didn't write a test around this: none of the other Task Listeners have a test.

I can also see that they are all basically the same and we could add some abstraction around them.

Please create a ticket in TechDebt, assign it to me and notify me, so I am aware of this repetition and the lack of tests and I can figure out a way to best do it. Once you created a ticket and notified me, mark this conversation as resolved.

Reply:
@filippo-possenti-version1 , Created ticket https://tools.hmcts.net/jira/browse/CIV-5794